op-challenger/game/fault/trace/split/provider.go

```diff
@@ -0,0 +1,83 @@
+package split
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/types"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var (
+	GetStepDataErr = fmt.Errorf("GetStepData not supported")
+	NoProvidersErr = fmt.Errorf("no trace providers configured")
+)
+
+var _ types.TraceProvider = (*SplitTraceProvider)(nil)
+
+// SplitTraceProvider is a [types.TraceProvider] implementation that
+// routes requests to the correct internal trace provider based on the
+// depth of the requested trace.
+type SplitTraceProvider struct {
+	logger     log.Logger
+	providers  []types.TraceProvider
+	depthTiers []uint64
```

Comment:
We don't need to make this support a generic number of providers - just two. We should keep it as simple as possible.

Reply:
Ok - that wasn't clear to me from [the issue description](https://github.com/ethereum-optimism/client-pod/issues/43) which detailed a generic split provider that routes based on depth